### DISCUSSION TITLE
"pdm export" hook in pre-commit gets skipped

### DISCUSSION THREAD

**Discussion by baojd42** [pdm-project/pdm#2957] [Q&A] [answered]
I add the pdm export hook in .pre-commit-config.yaml like,  
![image](https://github.com/pdm-project/pdm/assets/33564834/682c7da3-aa84-4a93-90e0-ea85426bdcd1)   .

But when I run pre-commit -a in the terminal, all hooks passed except that pdm-export-lock get skipped with message "no files to check".

![image](https://github.com/pdm-project/pdm/assets/33564834/b399ee12-e9d4-460b-8d8e-3654ac6e9168)

Directly calling "pdm export -o requirements.txt --without-hashes" in terminal works as expected.

I wonder what could be the reason. Should I provide more information for inspection?

Thanks.

**Comment by frostming** [comment, marked as answer]
It runs only when `pdm.lock` is changed in this commit

**Reply by baojd42** [reply]
Oh, I see. Thanks a lot.
